### PR TITLE
CNV-30298: Add UI for OVN Secondary Localnet Network

### PR DESCRIPTION
**Pull request by hstastna**
**Fixes**:
 https://issues.redhat.com/browse/CNV-30298

 **Solution Description**:
Add new _OVN Kubernetes secondary localnet network_ drop down item as another _Network Type_ in the Create NAD form. If the mentioned item is selected, display three additional fields:
- "Bridge Mapping"
- "MTU" (optional)
- "VLAN" (optional)

When the fields are filled in and _Create_ button clicked, create the new NAD with the config according to the newly added fields.

Add tooltip for "Bridge mapping" field:
_Physical network name. A bridge mapping must be configured on cluster nodes to map between physical network names and Open vSwitch bridges._

Additionally, use sentence case in the related page for fields' titles, such as "Resource name", "VLAN tag number", "IP address management", "Bridge name", "MAC spoof check". Also add missing translation support for those fields.

_Draft of the localnet documentation:_
https://github.com/openshift/openshift-docs/blob/0fe4ec0bcc31afd4ba5060fb87dc9c347da34603/modules/configuring-localnet-switched-topology.adoc#L92

_Some more doc related to the feature:_
https://github.com/openshift/openshift-docs/pull/61110

_Note:_
There isn't any related design doc for this feature, unfortunately.

**Screen shots / Gifs for design review**:
_Before:_
ovn_before

_After:_
New _OVN Kubernetes secondary localnet network_ drop down item added:
ovn1
ovn2
NAD creation, clicking on Create button with the new additional fields filled in:
ovn3
New NAD's config (yaml):
ovn4

### CODE REVIEW

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @upalatucci @vojtechszocs please review

File: frontend/packages/network-attachment-definition-plugin/src/constants/index.ts [hunk 1]
```diff
@@ -57,6 +68,21 @@ export const networkTypeParams: NetworkTypeParamsList = {
       initValue: true,
     },
   },
+  [ovnKubernetesSecondaryLocalnet]: {
+    bridgeMapping: {
+      name: 'Bridge Mapping',
```
Comment (pcbailey) [inline frontend/packages/network-attachment-definition-plugin/src/constants/index.ts:73]: These user-facing strings should be sentence case, not title case. It would be great if you could correct the others in this file as well.

Also, according to the Jira ticket the "Bridge mapping" label should have a tooltip explaining what it is. I looked for a design document to see if a tooltip was included, but didn't see a doc linked in the ticket.

**Review by pcbailey — COMMENTED** [review]
This is looking good! I just had two small comments.

File: frontend/packages/network-attachment-definition-plugin/src/constants/index.ts (hunk 1 above)
Comment (hstastna) [reply frontend/packages/network-attachment-definition-plugin/src/constants/index.ts:73]: Unfortunately, there isn't any design doc (this feature is not planned and prepared well, unfortunately, I asked for that but was told that it is not needed, I didn't agree but had no power to change that), there isn't any specific suggestion for the tooltip text (I asked for that but it seems nobody cares so probably it is not important anymore, so I wasn't trying to add it without any clear info because it doesn't worth the effort most of the time, it can be added later separately, when it will be specified and really needed) + I just followed the suggestions in the ticket, so I didn't use sentence case. The ticket says clearly about what they want to see in the UI. 

Anyway, the whole NADs section in the UI is very inconsistent with the rest of the UI, I am not going to fix this problem in this PR. If they will want it to be consistent, then I will use sentence case + will change all other related inconsistent texts, too. However, it seems nobody cares about consistency of NADs pages much, at least for now, except the two of us.
 
To sum up, I am not going to make these changes in this PR from various reasons, at least for now.

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @upalatucci @vojtechszocs please review

File: frontend/packages/network-attachment-definition-plugin/src/constants/index.ts (hunk 1 above)
Comment (pcbailey) [reply frontend/packages/network-attachment-definition-plugin/src/constants/index.ts:73]: I talked to Petr. He updated the casing to "Bridge mapping" and added the copy for the tooltip to the ticket. I asked Yifat and Avital to review it.

If you can update the case on "Bridge mapping" and add the tooltip once they finalize the copy I'll open a separate PR to update the remainder of the customer-facing strings since I'm sure I'm responsible for much of the inconsistency.

Comment (hstastna) [reply frontend/packages/network-attachment-definition-plugin/src/constants/index.ts:73]: Thanks. I'd wait for Avital to review it. She was pinged already longer time ago regarding this topic, but..

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @upalatucci @vojtechszocs please review

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @upalatucci @vojtechszocs please review

**Comment by hstastna** [comment]
Note that this PR needs the following labels:

/label docs-approved
/label px-approved
/label qe-approved

See the comment: https://github.com/openshift/console/pull/13291#issuecomment-1782825475
